LEAF_Request_Portal/admin/ajaxIndex.php

```diff
@@ -241,6 +249,14 @@ function checkToken()
         }
 
         break;
+
+    case 'gethistoryText':
+        $t_form = new Smarty;
```

Comment:
For ease in merging this with the stateless branch, please include:
`$t_form->setTemplateDir(__DIR__."/templates/")->setCompileDir(__DIR__."/templates_c/");`
These directories are being set relative to the path in the URL in the Smarty class, but they need to be specifically set for stateless to work.

Reply:
Done.